store/userStore.ts

```diff
@@ -1,5 +1,26 @@
 import { Language, UserInfo } from "@/types/user";
 import { create } from "zustand";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
+// Helper function to persist data
+const persistData = async (key: string, value: any) => {
+  try {
+    await AsyncStorage.setItem(key, JSON.stringify(value));
+  } catch (error) {
+    console.error("Error saving data", error);
+  }
+};
+
+// Helper function to load data
+const loadData = async (key: string) => {
```

Comment:
This is interesting to store each piece separately i.e. lang, user info etc. I would probably store it all in one go as a json object called userInfo instead of each k=>v